src/legacy/views/secondary-bar/parts/edit/share-folder-properties.tsx

```diff
@@ -3,55 +3,73 @@
  *
  * SPDX-License-Identifier: AGPL-3.0-only
  */
+import React, { FC, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
 import {
 	ButtonOld as Button,
 	Chip,
+	ChipProps,
 	Container,
 	Padding,
+	SnackbarManagerContext,
 	Text,
 	Tooltip
 } from '@zextras/carbonio-design-system';
-import { useUserAccounts } from '@zextras/carbonio-shell-ui';
-import { map, replace, split } from 'lodash';
-import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { Grant, soapFetch, useUserAccounts } from '@zextras/carbonio-shell-ui';
+import { map } from 'lodash';
 import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
-import { sendShareNotification } from '../../../../store/actions/send-share-notification';
-import { ShareFolderRoleOptions, findLabel } from '../../commons/utils';
-import { capitalise } from '../../utils';
+
 import { Context } from './edit-context';
 import { useAppDispatch } from '../../../../hooks/redux';
+import { sendShareNotification } from '../../../../store/actions/send-share-notification';
+import {
+	ActionProps,
+	GranteeInfoProps,
+	GranteeProps,
+	ShareFolderPropertiesProps
+} from '../../../../types/contact';
+import { ShareFolderRoleOptions, findLabel } from '../../commons/utils';
 
-const HoverChip = styled(Chip)`
-	background-color: ${({ theme, hovered }) =>
+const HoverChip = styled(Chip)<ChipProps & { hovered?: boolean }>`
```

Comment:
ChipProps type is not required. These props are already inferred by the styled function, which uses the Chip type props, plus the custom one that we define in the generic.